pkg/csilvm/metrics.go

```diff
@@ -0,0 +1,83 @@
+package csilvm
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/mesosphere/csilvm/pkg/lvm"
+	"github.com/uber-go/tally"
+	"google.golang.org/grpc"
+)
+
+func MetricsInterceptor(scope tally.Scope) grpc.UnaryServerInterceptor {
+	scope = scope.SubScope("requests")
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		scope = scope.Tagged(map[string]string{
+			"method": info.FullMethod,
+		})
+		timer := scope.Timer("duration")
```

Comment:
looks like tally has something nice and reusable https://github.com/uber-go/tally/blob/master/instrument/call.go

Reply:
Nice, i'll use that

Reply:
Hmm - that results in some awkward code and metrics called "/csi..." etc. I think I'll switch `duration` to `latency` and use `result_type: success/error`, but not switch to using their `instrument.Call` abstraction.